crates/net/network/src/transactions/validation.rs

```diff
@@ -0,0 +1,552 @@
+        if let Some(reasonable_max_encoded_tx_length) = self.max_encoded_tx_length(tx_type) {
+            if size > reasonable_max_encoded_tx_length {
+                debug!(target: "net::eth-wire",
+                    ty=ty,
+                    size=size,
+                    hash=%hash,
+                    reasonable_max_encoded_tx_length=reasonable_max_encoded_tx_length,
+                    strict_max_encoded_tx_length=self.strict_max_encoded_tx_length(tx_type),
+                    network=%Self,
+                    "tx size in eth68 announcement, is unreasonably large"
+                );
+
+                // just log a tx which is bigger than the reasonable max encoded tx length, don't
+                // filter it out
+            }
+        }
+
+        ValidationOutcome::Fetch
+    }
+
+    fn max_encoded_tx_length(&self, ty: TxType) -> Option<usize> {
+        // the biggest transaction so far is a blob transaction, which is currently max 2^17,
+        // encoded length, nonetheless, the blob tx may become bigger in the future.
+        match ty {
+            TxType::Legacy | TxType::EIP2930 | TxType::EIP1559 => {
+                Some(MAX_MESSAGE_SIZE)
+            }
+            TxType::EIP4844 => None,
+        }
+    }
+
+    fn strict_max_encoded_tx_length(&self, tx: TxType) -> Option<usize> {
+        None
+    }
+
+    fn min_encoded_tx_length(&self, _ty: TxType) -> Option<usize> {
+        // a transaction will have at least a signature. the encoded signature encoded on the tx
+        // is at least as big as the decoded type.
+        Some(SIGNATURE_DECODED_SIZE_BYTES)
+    }
+
+    fn strict_min_encoded_tx_length(&self, tx: TxType) -> Option<usize> {
+        // decoding a tx will exit right away if it's not at least a byte
+        Some(1)
+    }
+}
+
+impl FilterAnnouncement for EthAnnouncementFilter {
+    fn filter_valid_entries_68(
+        &self,
+        msg: NewPooledTransactionHashes68,
+    ) -> (FilterOutcome, ValidAnnouncementData)
+    where
+        Self: ValidateTx68,
+    {
+        trace!(target: "net::tx::validation",
+            types=?msg.types,
+            sizes=?msg.sizes,
+            hashes=?msg.hashes,
+            network=%Self,
+            "validating eth68 announcement data.."
+        );
+
+        let NewPooledTransactionHashes68 { mut hashes, mut types, mut sizes } = msg;
+
+        debug_assert!(
+            hashes.len() == types.len() && hashes.len() == sizes.len(), "`%hashes`, `%types` and `%sizes` should all be the same length, decoding of `NewPooledTransactionHashes68` should handle this, 
+`%hashes`: {hashes:?}, 
+`%types`: {types:?}, 
+`%sizes: {sizes:?}`"
+        );
+
+        //
+        // 1. checks if the announcement is empty
+        //
+        if hashes.is_empty() {
+            debug!(target: "net::tx",
+                network=%Self,
+                "empty eth68 announcement"
+            );
```

Comment:
hmm, this should work since the filter is a ZST, which means the display impl won't be too noisy when running in debug mode

Reply:
added a test for it to make sure